### PR TITLE
Add manual job for rebuilding reference docs

### PR DESCRIPTION
**Pull request by bradfordcp**
**What this PR does**:
Adds a manual Github CI action for rebuilding the reference docs. This alleviates the protected branch issue we're currently experiencing on main

**Checklist**
- [x] Changes manually tested
- [x] Chart versions updated (if necessary)
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by bradfordcp** [comment]
This PR is missing a modified exit command. We want to allow for the push command to fail if there are no changes.

**Comment by bradfordcp** [comment]
After merging into the main branch a maintainer may visit the "actions" link at the top of the repo. A new "Regenerate Reference Docs" job is visible.

ci-trigger

When clicking the "Run workflow" button a dialog is shown listing available branches

ci-trigger-2

Clicking the "Run" button then schedules the job on the branch.

ci-queued-3

The job runs and output may be viewed (like other actions)

regenerated-docs-1

From here we can look back at the PR and see the generated docs commit

docs-commit

Note this task will NOT work on the `main` branch as it is protected and the push will fail.

**Review by jsanda — APPROVED** [review]

**Review by jdonenine — APPROVED** [review]

**Pull request merged**
